### ISSUE TITLE
Engine warning and timeout

### ISSUE THREAD

**Issue by gppleton-platform** [balmasi/migrate-mongoose#83]
```
$ migrate up
(node:611) [MONGODB DRIVER] Warning: Current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
Synchronizing database with file system migrations...
Failed to run migration AssetTableAddTypeField due to an error.
Not continuing. Make sure your data is in consistent state
Operation `rooms.find()` buffering timed out after 10000ms
```

This is my first time trying to use this library. Since we don't handle the connection ourselves I'm not sure how to fix the warning. And I have no idea as to why my simple find() operation is timing out. The migration table was created just fine within the collection.

I've set MIGRATE_dbConnectionUri=mongodb://mongodb:27017/diageo

**Comment by slampazona** [comment]
Same error here ... And the "migration" collection is created so the connection is OK

**Comment by dominikmch** [comment]
Had the same issue today. It was because migrate-mongoose script wanted to access schemas before it connected to the DB. For me, adding mongoose.connect(MONGO_URI); at top of the migration script helped.

**Comment by slampazona** [comment]
Yep @dominikmch , i have used this same workaround. But i must add it on **every** migration scripts :/

Like this : 
```js
async function up() {
  await connectDB();
  await asyncForEach(users, async (user) => {
    await User.create({
      ...user,
      token: token(),
      origin: 'migrationScript',
    });
  });
}

/**
 * Make any changes that UNDO the up function side effects here (if possible)
 */
async function down() {
  await connectDB();
  // Write migration here
  await User.deleteMany({ origin: 'migrationScript' });
}
```

I noticed that only happen with **import** use, i have antoher project with only **require** and i dont have this issue.

**Comment by farmergaetano** [comment]
@balmasi, could we add the `useUnifiedTopology: true` connection option in the `Migrator` constructor (in `src/lib.js`) as per [this answer](https://stackoverflow.com/a/57899638)? I've tested locally and it removes the warning.

**Comment by moonninja** [comment]
This happens when you pull User directly from you user.model.js file, use the index.js example as in the setup documentation and pull User from there, this makes a db connection before trying to do anything. Will work


`const {User} = require('../database/models/index.js')

/**
 * Make any changes you need to make to the database here
 */
async function up () {
  await User.create({ firstName: 'user1'});
  await User.create({ firstName: 'user2'});
}

/**
 * Make any changes that UNDO the up function side effects here (if possible)
 */
async function down () {

}

module.exports = { up, down };`

**Comment by mariuspopaadrian** [comment]
> @balmasi, could we add the `useUnifiedTopology: true` connection option in the `Migrator` constructor (in `src/lib.js`) as per [this answer](https://stackoverflow.com/a/57899638)? I've tested locally and it removes the warning.

useUnifiedTopology  seems to be deprecated https://www.mongodb.com/community/forums/t/mongodb-nodejs-driver-6-0-0-released/241691#usenewurlparser-and-useunifiedtopology-emit-deprecation-warnings-16
